### PR TITLE
Document the push flow and why the proxy is nece…

### PR DESCRIPTION
**Pull request by nickvergessen**
…ssary

Signed-off-by: Joas Schilling <coding@schilljs.com>

### CODE REVIEW

File: docs/push-v2.md [hunk 1]
```diff
@@ -1,5 +1,55 @@
 # Push notifications as a Nextcloud client device
 
+## Introduction
+
+> Why is push-notifications.nextcloud.com necessary?
+
+The mobile apps people install from the Google Playstore and Apple App Store are signed with Nextcloud developer keys or certificates.
+Push notifications sent to those devices need to be signed with a generated push key or certificate from the same developer account.
+The keys and certificates can not be shipped with the Nextcloud server as otherwise everyone would have our developer key and could manipulate releases or push to any random Nextcloud device.
+The Firebase Cloud Messaging (Google) and Apple Push Notification Service are not made for something like a federated produce like Nextcloud and still assume there is a single entity behind them like with all the other services.
```
Comment (SystemKeeper) [inline docs/push-v2.md:10]:
```suggestion
The Firebase Cloud Messaging (Google) and Apple Push Notification Service are not made for something like a federated product like Nextcloud and still assume there is a single entity behind them like with all the other services.
```

File: docs/push-v2.md [hunk 2]
```diff
@@ -1,5 +1,55 @@
 # Push notifications as a Nextcloud client device
 
+## Introduction
+
+> Why is push-notifications.nextcloud.com necessary?
+
+The mobile apps people install from the Google Playstore and Apple App Store are signed with Nextcloud developer keys or certificates.
```
Comment (szaimen) [inline docs/push-v2.md:7]:
```suggestion
The Talk mobile apps from the Google Playstore and Apple App Store are signed with Nextcloud developer keys or certificates.
```

**Review by SystemKeeper — CHANGES_REQUESTED** [review]

File: docs/push-v2.md (hunk 1 above)
Comment (szaimen) [reply docs/push-v2.md:10]: Or rather project?

File: docs/push-v2.md (hunk 2 above)
Comment (SystemKeeper) [reply docs/push-v2.md:7]: The push proxy is used for the files apps as well.

File: docs/push-v2.md (hunk 1 above)
Comment (SystemKeeper) [reply docs/push-v2.md:10]: project sounds good to me as well

File: docs/push-v2.md (hunk 2 above)
Comment (szaimen) [reply docs/push-v2.md:7]: Then maybe simply Nextcloud mobile apps?

**Review by szaimen — APPROVED** [review]
LGTM except the details

**Review by SystemKeeper — APPROVED** [review]

**Pull request merged**